test/flush_worker_test.cc

```diff
@@ -47,7 +48,10 @@ class FlushWorkerTest : public Test {
         .WillRepeatedly(Invoke([&](const std::chrono::microseconds,
                                    const Envoy::ScopeTrackedObject*) { timer_set_ = true; }));
     EXPECT_CALL(*timer_, disableTimer()).WillOnce(Invoke([&]() { timer_set_ = false; }));
-    EXPECT_CALL(*dispatcher_, exit()).WillOnce(Invoke([&]() { stopped_ = true; }));
+    EXPECT_CALL(*dispatcher_, exit()).WillOnce(Invoke([&]() {
+      absl::MutexLock l(&lock_);
```

Comment:
Sorry, I should have thought of this earlier, but thoughout the code base we've been using `Envoy::Thread::LockGuard` [1] in favor of `absl::MutexLock`. I bet one is implemented in terms of the other, but for consistency it might be good to stick to `Envoy::Thread::LockGuard`?

[1] https://github.com/envoyproxy/envoy/blob/7e88243610c09a8346226b49528bc7516fcfd7a7/source/common/common/lock_guard.h#L95

Reply:
Thanks Otto. Done